packages/diagrams/backend/sirius-components-collaborative-diagrams/src/main/java/org/eclipse/sirius/components/collaborative/diagrams/services/ParametricSVGImageFactory.java

```diff
@@ -0,0 +1,202 @@
+            .flatMap(service-> service.getImages().stream())
+            .filter(image -> {
+                return svgName.equals(image.getId().toString());
+            })
+            .findFirst();
+        // @formatter:on
+
+        if (svgImageOpt.isPresent()) {
+            ClassPathResource classPathResource = new ClassPathResource(svgImageOpt.get().getPath());
+            try (InputStream inputStream = classPathResource.getInputStream()) {
+                DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+                factory.setFeature(XMLConstants.FEATURE_SECURE_PROCESSING, true);
+                factory.setFeature("http://apache.org/xml/features/disallow-doctype-decl", true); //$NON-NLS-1$
+                Document document = factory.newDocumentBuilder().parse(inputStream);
+                XPath xpath = XPathFactory.newInstance().newXPath();
+
+                // change the global size
+                Node svgNode = ((NodeList) xpath.evaluate("/svg", document, XPathConstants.NODESET)).item(0); //$NON-NLS-1$
+                if (svgNode instanceof Element) {
+                    Element element = (Element) svgNode;
+                    Double halfBorderSize = Double.valueOf(attributesValues.get(SVGAttribute.BORDERSIZE)) * 0.5d;
+                    String realWidth = String.valueOf(Double.valueOf(attributesValues.get(SVGAttribute.WIDTH)) + halfBorderSize * 2);
+                    String realHeight = String.valueOf(Double.valueOf(attributesValues.get(SVGAttribute.HEIGHT)) + halfBorderSize * 2);
+                    String viewBox = String.format(Locale.US, "-%f -%f %s %s", halfBorderSize, halfBorderSize, realWidth, realHeight); //$NON-NLS-1$
+                    element.setAttribute(WIDTH, realWidth);
+                    element.setAttribute(HEIGHT, realHeight);
+                    element.setAttribute("viewBox", viewBox); //$NON-NLS-1$
+
+                }
+
+                String expr = String.format("//*[contains(@id, '%s')]|//*[contains(@id, '%s')]", RECTANGLE_ELEMENT_LABEL_ID, RECTANGLE_ELEMENT_MAIN_ID); //$NON-NLS-1$
+                NodeList nodes = (NodeList) xpath.evaluate(expr, document, XPathConstants.NODESET);
+
+                for (int i = 0; i < nodes.getLength(); i++) {
+                    Element node = (Element) nodes.item(i);
+
+                    this.updateNode(attributesValues, node, svgImageOpt.get());
+                }
+
+                ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+                Source xmlSource = new DOMSource(document);
+                Result outputTarget = new StreamResult(outputStream);
+                try {
+                    TransformerFactory.newInstance().newTransformer().transform(xmlSource, outputTarget);
+                } catch (TransformerException | TransformerFactoryConfigurationError e) {
+                    this.logger.error(e.getMessage());
```

Comment:
`logger.warn`, we keep `logger.error` for major Spring errors

Reply:
done